pkg/process/utils.go

```diff
@@ -137,6 +137,8 @@ func checkKillError(err error) error {
 		strings.Contains(strings.ToLower(err.Error()), "no such process") ||
 		err == unix.ESRCH {
 		return errors.Wrapf(errdefs.ErrNotFound, "process already finished")
+	} else if strings.Contains(err.Error(), "does not exist") {
+		return errors.Wrapf(errdefs.ErrNotFound, "no such container")
```

Comment:
Change looks good. Could you add integration test case here? Thanks!

Reply:
Of course.

Reply:
After I studied two hours, I found it impossible to create a scene, where containerd-shim deleted failed and container deleted by runc, through a integration test case. Maybe this is my problem, I don't know how to do with containerd client.

Reply:
Probably you'll need to manually delete the container from runc?

Reply:
> Probably you'll need to manually delete the container from runc?
Sure, or make umount fail by some tricks.

Reply:
```
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/containerd/containerd"
	"github.com/containerd/containerd/cio"
	"github.com/containerd/containerd/namespaces"
	"github.com/containerd/containerd/oci"
	"github.com/containerd/containerd/runtime/v2/runc/options"
	"github.com/containerd/go-runc"
)

func main() {
	client, err := containerd.New("/run/containerd/containerd.sock")
	if err != nil {
		log.Fatal(err)
	}
	defer client.Close()

	ctx := namespaces.WithNamespace(context.Background(), "example")
	image, err := client.Pull(ctx, "docker.io/library/busybox:latest", containerd.WithPullUnpack)
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("Successfully pulled %s image\n", image.Name())

	runtimeOpts := &options.Options{
		Root: "/tmp/runc",
	}
	container, err := client.NewContainer(
		ctx,
		"sleep-busybox",
		containerd.WithNewSnapshot("sleep-busybox", image),
		containerd.WithNewSpec(oci.WithImageConfig(image), oci.WithProcessArgs("sleep", "1d")),
		containerd.WithRuntime("io.containerd.runc.v2", runtimeOpts),
	)
	if err != nil {
		log.Fatal(err)
	}
	defer container.Delete(ctx, containerd.WithSnapshotCleanup)

	task, err := container.NewTask(ctx, cio.NewCreator(cio.WithStdio))
	if err != nil {
		log.Fatal(err)
	}
	defer func() {
		if _, err := task.Delete(ctx, containerd.WithProcessKill); err != nil {
			log.Fatal(err)
		}
	}()

	exitStatusC, err := task.Wait(ctx)
	if err != nil {
		log.Fatal(err)
	}

	if err := task.Start(ctx); err != nil {
		log.Fatal(err)
	}

	select {
	case <-exitStatusC:
	default:
		rcmd := &runc.Runc{
			Root: "/tmp/runc/example/",
		}

		if err := rcmd.Delete(ctx, "sleep-busybox", &runc.DeleteOpts{Force: true}); err != nil {
			log.Fatal(err)
		}
		fmt.Println(<-exitStatusC)
	}
}
```

How about this?

Reply:
Most codes are from https://containerd.io/docs/getting-started/ (very very helpful). Just added runc root when create container. BTW